components/search/SearchHistory.vue

```diff
@@ -0,0 +1,57 @@
+<script setup lang="ts">
+import type { SavedSearchResult } from '@/composables/masto/search'
+
+const props = defineProps<{
+  searchHistory: SavedSearchResult[]
+}>()
+
+const emit = defineEmits<{
+  (event: 'clear-all'): void
+  (event: 'clear-history-item', id: string): void
+}>()
+
+const orderedHistoryItems = computed(() => {
+  return [...props.searchHistory].sort((a, b) => new Date(b.lastQueryTime).getTime() - new Date(a.lastQueryTime).getTime())
+})
+</script>
+
+<template>
+  <div left-0 top-12 w-full group-focus-within="pointer-events-auto visible" pointer-events-none>
+    <div class="grid grid-cols-[1fr_auto] place-items-center mx-2 mb-2">
+      <div justify-self-start text-xl font-500>
+        {{ $t('search.search_history_title') }}
+      </div>
+      <button
+        gap-1 items-center
+        border-1
+        rounded-full flex="~ gap2 center" font-400 px3 py1
+        bg-base border-primary text-primary
+        hover="text-inverted bg-primary border-primary"
+        :aria-label="$t('search.search_clear_all')"
+        @click="emit('clear-all')"
+      >
+        <span>{{ $t('search.search_clear_all') }}</span>
+      </button>
```

Comment:
The click to clear all button and the trash can icon don't seem to be working properly when being clicked, the history is not being cleared.

Reply:
@LittleSound , could you be more specific? Have you tried a local build? Using the [deploy preview](https://deploy-preview-1639--elk-zone.netlify.app/) above it works as expected. I add here two hashtags, they do appear in the history, then with Clear all they are removed:


![elk](https://user-images.githubusercontent.com/7863452/218311322-69a9a280-0503-4cc8-b7d7-51cd5a32be44.gif)

Reply:
Maybe it's because I'm testing on a mobile device and you're testing on a computer, and my browser is Safari by the way.

https://user-images.githubusercontent.com/19204772/218327338-5e814c57-ff94-4041-aed7-5a262c2280b0.MOV

Reply:
Hmm. We need to investigate further. It works fine on Android both with Chrome and even with Samsung Internet browser:

https://user-images.githubusercontent.com/7863452/218332319-e9411ac8-7e59-46e1-97f3-3474da48aa00.mp4


https://user-images.githubusercontent.com/7863452/218332324-f789fd19-c58a-47ff-b977-cc2b59d866bc.mp4